### PR TITLE
Add pull request option support

### PR DESCRIPTION
**Pull request by chalenge**
Closes #

## About
Add ability to pass the pull request number associated with CI/CD event commit

### CODE REVIEW

File: faros_event.sh [hunk 1]
```diff
@@ -90,6 +90,7 @@ function help() {
     echo "Argument                | Req | Allowed Values / URI form"
     echo "-----------------------------------------------------------------------------"
     echo "--commit                | Yes | $commit_uri_form"
+    echo "--pull_request_number   |     | e.g. 1 (should be a number)"
```
Comment (tovbinm) [inline faros_event.sh:93]: `e.g. 123`

File: faros_event.sh [hunk 2]
```diff
@@ -107,20 +108,22 @@ function help() {
     echo "--deploy_status         | Yes | $deploy_statuses"
     echo "--artifact              | *2  | $artifact_uri_form"
     echo "--commit                | *2  | $commit_uri_form"
+    echo "--pull_request_number   |     | e.g. 1 (should be a number) *3"
```
Comment (tovbinm) [inline faros_event.sh:111]: I think it should be `| *3 | e.g. 123 (should be a number)`

File: test/demo.sh [hunk 3]
```diff
@@ -23,10 +23,23 @@ export FAROS_DRY_RUN=1 # Set to 0 or comment out to send the event to Faros
 # ../faros_event.sh CI \
```
Comment (tovbinm) [inline test/demo.sh:23]: why all the tests are commented out?!

Comment (tovbinm) [reply test/demo.sh:23]: or these are not tests? what are these for?!

Comment (chalenge) [reply test/demo.sh:23]: example commands one can run. Not sure if they should be in the test directory

Comment (willmarks) [reply test/demo.sh:23]: I committed them for ease of manual testing and/or demoing the CLI. Im not attached to them if we would rather remove them... also fine with putting them in another place.

File: faros_event.sh [hunk 4]
```diff
@@ -107,20 +108,22 @@ function help() {
     echo "--deploy_status         | Yes | $deploy_statuses"
     echo "--artifact              | *2  | $artifact_uri_form"
     echo "--commit                | *2  | $commit_uri_form"
+    echo "--pull_request_number   |     | *3 | e.g. 123 (should be a number)"
```
Comment (willmarks) [inline faros_event.sh:111]: I think we want `*3` to be in `Req` column

Comment (chalenge) [reply faros_event.sh:111]: But it is never required, so maybe `--pull_request_number   |     | *3 e.g. 123 (should be a number)"

Comment (tovbinm) [reply faros_event.sh:111]: +1 @willmarks  it's  conditionally required, that's whye there is the `*`

File: README.md [hunk 5]
```diff
@@ -101,6 +101,7 @@ A `CI` event communicates the outcome of a code build pipeline execution, and it
 | Argument                           | Description                                                                                                                                                   | Required                        | Allowed Value                                               |
 | ---------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------- | ----------------------------------------------------------- |
 | &#x2011;&#x2011;commit             | The URI of the commit. (`<source>://<organization>/<repository>/<commit_sha>` e.g. `GitHub://faros-ai/my-repo/da500aa4f54cbf8f3eb47a1dc2c136715c9197b9`)      | Yes                             |                                                             |
+| &#x2011;&#x2011;pull_request_number | The pull request number of the commit. (`e.g. 1`)       |                                 |                                                             |
```
Comment (tovbinm) [inline README.md:104]: e.g. 123

File: README.md [hunk 6]
```diff
@@ -123,6 +124,7 @@ A `CD` event communicates the outcome of an application deployment pipeline exec
...
 | &#x2011;&#x2011;artifact              | The URI of the artifact. (`<source>://<organization>/<repository>/<artifact_id>` e.g. `DockerHub://farosai/my-repo/da500aa4f54cbf8f3eb47a1dc2c136715c9197b9`) | Either &#x2011;&#x2011;commit or &#x2011;&#x2011;artifact |                                                                |
 | &#x2011;&#x2011;commit                | The URI of the commit. (`<source>://<organization>/<repository>/<commit_sha>` e.g. `GitHub://faros-ai/my-repo/da500aa4f54cbf8f3eb47a1dc2c136715c9197b9`)      | Either &#x2011;&#x2011;commit or &#x2011;&#x2011;artifact |                                                                |
+| &#x2011;&#x2011;pull_request_number | The pull request number of the commit. (`e.g. 1`). Used only if --commit is included       |                                 |                                                             |
```
Comment (tovbinm) [inline README.md:127]: e.g. 123 as well

File: test/spec/faros_event_spec.sh [hunk 7]
```diff
@@ -193,6 +244,39 @@ Describe 'faros_event.sh'
...
+
+    CIWithPullRequestExpectedOutput='{"type":"CI","version":"0.0.1","origin":"Faros_Script_Event","data":{"artifact":{"id":"<artifact>","repository":"<artifact_repo>","organization":"<artifact_org>","source":"<artifact_source>"},"commit":{"sha":"<commit_sha>","repository":"<vcs_repo>","organization":"<vcs_organization>","source":"<vcs_source>","pullRequestNumber":101},"run":{"id":"<build_uid>","pipeline":"<cicd_pipeline>","organization":"<cicd_organization>","source":"<cicd_source>","status":"Success"}}}'
+
+    It 'constructs correct event when pull request included'
+      ci_event_test() {
+        echo $(
+          ../faros_event.sh CI -k "<api_key>" \
+          --artifact "<artifact_source>://<artifact_org>/<artifact_repo>/<artifact>" \
+          --commit "<vcs_source>://<vcs_organization>/<vcs_repo>/<commit_sha>" \
+          --pull_request_number "101" \
```
Comment (tovbinm) [inline test/spec/faros_event_spec.sh:256]: can we allow `--pull_request_number 101` ?

Comment (chalenge) [reply test/spec/faros_event_spec.sh:256]: Yes, it is allowed. Updated the README and added test

**Review by willmarks — APPROVED** [review]

**Comment by tovbinm** [comment]
nice!

bump the version + release

**Pull request merged**